### PR TITLE
Adding Pre-Trained Autoencoder

### PR DESCRIPTION
**Pull request by Maxusmusti**
Also fixing minor bugs, formatting, and import errors found

### CODE REVIEW

File: trainer.py [hunk 1]
```diff
@@ -7,18 +7,29 @@
...
+from transformers import ViTMAEForPreTraining
 from data import get_train_loader, get_val_loader
+from diffusion import Diffusion
+from unet import UNet
 
 class Trainer:
     def __init__(self, args):
         self.args = args
         self.device = args.device
         self.train_loader = get_train_dataloader(args)
```
Comment (jjlee0802cu) [inline trainer.py:19]: Why are the data imports and the function called here different?

Comment (Maxusmusti) [reply trainer.py:19]: Because you wrote it like that :)

Comment (jjlee0802cu) [reply trainer.py:19]: Will add train, val, and test data loader methods to the data module in the future. This is fine for now.

**Review by jjlee0802cu — APPROVED** [review]
LGTM

**Pull request merged**